tests/layers/hunyuandit_attn_test.py

```diff
@@ -70,36 +70,55 @@ def setUp(self):
             p2.data.copy_(p1.data)
 
         get_cache_manager().register_cache_entry(
-            self.attn2, layer_type="attn", cache_type="naive_cache"
+            self.attn2, layer_type="attn", cache_type="sequence_parallel_attn_cache"
         )
 
     def test_hunyuandit_attn(self):
         torch.manual_seed(0)
 
-        norm_hidden_states = torch.randn(1, 20, 32, dtype=self.dtype).cuda(
+        # prepare inputs
+        norm_hidden_states = torch.randn(
+            1, self.sequence_len, self.dim, dtype=self.dtype
+        ).cuda(self.local_rank)
+        dist.broadcast(norm_hidden_states, 0)
+
+        use_image_rotary_emb = False
+
+        # prepare rope parameters
+        cos = torch.randn(self.sequence_len, self.hidden_dim, dtype=self.dtype).cuda(
             self.local_rank
         )
-        image_rotary_emb = torch.randn(1, 20, 32, dtype=self.dtype).cuda(
+        dist.broadcast(cos, 0)
+        sin = torch.randn(self.sequence_len, self.hidden_dim, dtype=self.dtype).cuda(
             self.local_rank
         )
-
-        attn_output1 = self.attn1(norm_hidden_states, image_rotary_emb)
+        dist.broadcast(sin, 0)
+        image_rotary_emb = cos, sin
+        cos_shard = torch.chunk(cos, self.world_size, dim=0)[self.local_rank]
+        sin_shared = torch.chunk(sin, self.world_size, dim=0)[self.local_rank]
+        image_rotary_emb_shard = cos_shard, sin_shared
+
+        attn_output1 = self.attn1(
+            norm_hidden_states,
+            encoder_hidden_states=None,
```

Comment:
The test will fail.

encoder_hidden_states=norm_hidden_states
can pass unitest.